newIDE/app/src/EventsSheet/InstructionEditor/EventTextDialog.js

```diff
@@ -0,0 +1,240 @@
+// @flow
+import { Trans } from '@lingui/macro';
+import { t } from '@lingui/macro';
+
+import * as React from 'react';
+import Dialog from '../../UI/Dialog';
+import FlatButton from '../../UI/FlatButton';
+import HelpButton from '../../UI/HelpButton';
+import { Line, Column } from '../../UI/Grid';
+import ColorPicker from '../../UI/ColorField/ColorPicker';
+import MiniToolbar, { MiniToolbarText } from '../../UI/MiniToolbar';
+import SemiControlledTextField from '../../UI/SemiControlledTextField';
+
+const gd = global.gd;
+
+const toolbarItemStyle = {
+  marginRight: 10,
+};
+
+const styles = {
+  sizeTextField: {
+    width: 90,
+    ...toolbarItemStyle,
+  },
+  toolbarItem: toolbarItemStyle,
+  checkbox: toolbarItemStyle,
+};
+type RGBColor = {|
+  r?: number,
+  g?: number,
+  b?: number,
+  a?: number,
+|};
+
+type Props = {|
+  event: gdBaseEvent,
+  onClose: () => void,
+  onApply: () => void,
+|};
+
+type State = {|
+  textValue: string,
+  textColor: RGBColor,
+  backgroundColor: RGBColor,
+|};
+
+export default class EventTextDialog extends React.Component<Props, State> {
+  _applyChangesOnEvent = () => {
+    const { event } = this.props;
+    const { textValue, textColor, backgroundColor } = this.state;
+    const eventType = event.getType();
+
+    if (eventType === 'BuiltinCommonInstructions::Comment') {
+      //Text value
+      gd.asCommentEvent(event).setComment(textValue);
+
+      //Text color
+      gd.asCommentEvent(event).setTextColor(
+        parseInt(textColor.r),
```

Comment:
Always always always specify the second argument to parseInt: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt

```suggestion
        parseInt(textColor.r, 10),
```

Reply:
About parseInt(), we can ignore it here ?
backgroundColor is define with getBackgroundColorXXX() and this return always a number. And ColorPicker return number too.

Unless it is necessary to do this check because the values of the.json file can be edited and falsified.

I'm wondering when it's judicious to check the type and format of the data when the methods used do already a checking.
Any way I keep it, it dosen't cost much.

Reply:
Wait a second you're right we should always have number (if we have strings there, it's a big problem somewhere).
So you can remove all the parseInt entirely: `gd.asCommentEvent(event).setTextColor(textColor.r, textColor.g, textColor.b)` :)

> Unless it is necessary to do this check because the values of the.json file can be edited and falsified.

No you can remove these parseInt, even if you go and replace them by a string in the JSON, it's being read by libGD.js which will probably read it as 0 or something like this. 
In either case, methods from libGD.js being in C++, they are guaranteed to always return a number if they declare to return a float/double/int in `Bindings.idl`. (and they also expect to receive arguments with the exact type declared in `Bindings.idl`, so if for some reason a string is sent to a function expecting a number, very bad things can happen).

> I'm wondering when it's judicious to check the type and format of the data when the methods used do already a checking.

The idea with Flow is to get the guarantee of the type and format of data. A bit more painful to write at first, but it's actually leading to cleaner code because you don't need extra check/extra conversion/extra handling of undefined values :)